service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -110,11 +110,21 @@ private RunSetDetailsResponse convertToRunSetDetails(RunSet runSet) {
   public ResponseEntity<RunSetListResponse> getRunSets(UUID methodId, Integer pageSize) {
     List<RunSet> runSets = runSetDao.getRunSets();
     List<RunSet> updatedRunSets = smartRunSetsPoller.updateRunSets(runSets);
-
-    List<RunSetDetailsResponse> runSetDetails =
-        updatedRunSets.stream().map(this::convertToRunSetDetails).toList();
-    RunSetListResponse response = new RunSetListResponse().runSets(runSetDetails);
-
+    RunSetListResponse response;
+
+    if (methodId != null) {
```

Comment:
should it be possible for `methodId` to be null?

genuinely asking - I don't know if this is a pattern that we use/should use in this codebase. do we already use this method somewhere where `methodId` is null?

if we are willing to allow `methodId` to be null, then the fallback behavior you have here makes sense

Reply:
Well, if we don't supply a `methodId`, then doesn't it get passed into CBAS as a null value?

Reply:
yeah that sounds right. I was less questioning whether or not this will run as expected (I think you're right that it will) and more questioning whether we want to design methods to behave this way when arguments aren't supplied.

thinking about it ... this is more "food for thought" than something that should block acceptance. I'll be curious to hear @salonishah11 's or @cjllanwarne 's opinion

Reply:
Short answer: yes. This is how the spring/openapi libraries collaborate to implement optional parameters on APIs.

You might ask for the sense in allowing users to ask for _all run sets_ all at the same time... but in terms of implementing the API as currently defined, it is correct